pkg/cli/publish.go

```diff
@@ -71,17 +78,53 @@ func PublishCmd(ctx context.Context, outputRefs string, opts ...build.Option) er
 		return err
 	}
 
+	if len(archs) == 0 {
+		archs = types.AllArchs
+	}
+	if len(bc.ImageConfiguration.Archs) == 0 {
+		bc.ImageConfiguration.Archs = archs
+	}
+
 	log.Printf("building tags %v", bc.Tags)
 
-	layerTarGZ, err := bc.BuildLayer()
-	if err != nil {
-		return fmt.Errorf("failed to build layer image: %w", err)
-	}
-	defer os.Remove(layerTarGZ)
+	var digest name.Digest
+	switch len(bc.ImageConfiguration.Archs) {
+	case 0:
+		return fmt.Errorf("no archs requested: %w", err)
+	case 1:
+		bc.Arch = bc.ImageConfiguration.Archs[0]
+		layerTarGZ, err := bc.BuildLayer()
+		if err != nil {
+			return fmt.Errorf("failed to build layer image: %w", err)
+		}
+		defer os.Remove(layerTarGZ)
 
-	digest, err := oci.PublishImageFromLayer(layerTarGZ, bc.ImageConfiguration, bc.SourceDateEpoch, bc.Tags...)
-	if err != nil {
-		return fmt.Errorf("failed to build OCI image: %w", err)
+		digest, _, err = oci.PublishImageFromLayer(layerTarGZ, bc.ImageConfiguration, bc.SourceDateEpoch, bc.Arch, bc.Tags...)
+		if err != nil {
+			return fmt.Errorf("failed to build OCI image: %w", err)
+		}
+	default:
+		var imgs []v1.Image
+		workDir := bc.WorkDir
+		for _, arch := range archs {
```

Comment:
We should open an issue to track doing these in parallel (and track any hurdles that might have to clear)

Reply:
```
2022/03/08 15:48:08 error during command execution: failed to build layer image for "386": failed to initialize apk database: failed to run /sbin/apk add --initdb --arch aarch64 --root /tmp/apko-4213008895/s390x: exit status 99
```

Why is it using s390x root for aarch64?  Is there an issue with the current code?

Reply:
You could let `apk` wait for lock with `--wait`

Reply:
> Why is it using s390x root for aarch64? Is there an issue with the current code?

Good catch. I don't believe there's a bug in the current code. I'm pretty sure this was coming from how terribly lazy I was being in adding concurrency -- since each iteration of the loop sets `bc.Arch = arch`, this was shared between concurrent builds (and wouldn't be for sequential builds), which might have led to this bug.

In any case, I'll be sure to do a less lazy thing when we actually make these run concurrently.

Reply:
alright, i'll test it in a bit and merge it :)